src/interp.rs

```diff
@@ -261,12 +299,46 @@ impl<'a, 'tcx> Machine<'a, 'tcx> {
     /// location inside the Machine.
     fn eval_place(&mut self, place: &Place<'tcx>) -> Address {
         match *place {
-            Place::Local(local) => Address::Local(local),
+            Place::Local(local) => Address::Local(local, 0),
             Place::Static(ref static_) => unimplemented!(),
-            Place::Projection(ref proj) => unimplemented!(),
+            Place::Projection(ref proj) => {
+                match proj.elem {
+                    ProjectionElem::Field(field, _) => {
+                        let base_ty = self.place_ty(&proj.base);
+                        let offset = self.get_field_offset(base_ty, field.index());
+                        match proj.base {
+                            Place::Local(local) => {
+                                return Address::Local(local, offset);
+                            },
+                            _ => unimplemented!()
+                        }
+                    }
+                    _ => unimplemented!("{:?}", proj.elem)
+                }
+            }
+        }
+    }
+
+    fn get_field_offset(&self, ty: Ty<'tcx>, field: usize) -> usize {
```

Comment:
The function is called `get_field_offset`, but returns a value named `size`. Should one be renamed?

Reply:
Perhaps it is misleading. The size is just a running total of the type size of preceding indices, which is then used to workout at what offset I should start reading from.

Reply:
Why not call it `offset`?

Reply:
Yeah, that has been changed now

Reply:
Thanks